chromium_src/chrome/browser/ui/tabs/tab_features.cc

```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/tabs/public/tab_features.h"
+
+#include "brave/browser/ui/side_panel/brave_side_panel_utils.h"
+
+#define Init Init_ChromiumImpl
+#include "src/chrome/browser/ui/tabs/tab_features.cc"
+#undef Init
+
+namespace tabs {
+
+void TabFeatures::Init(TabInterface& tab, Profile* profile) {
+  Init_ChromiumImpl(tab, profile);
+  // Expect upstream's Init to create the registry.
+  CHECK(side_panel_registry_.get());
+  brave::RegisterContextualSidePanel(side_panel_registry_.get(),
```

Comment:
why is this happening in TabFeatures instead of `SidePanelRegistry`?

Reply:
Upstream instantiates `SidePanelRegistry` in 2 ways:
 ```
After this CL, SidePanelCoordinator owns the window-scoped instance of the registry,
and TabFeatures owns the tab-scoped instance of the registry.
```
I believe we only want to plug into the tab-scoped instances.

Reply:
what I'm saying is that TabFeatures already creates an instance of SidePanelRegistry so why are we doing this registration outside of that?

Reply:
where does this happen in upstream?

Reply:
It looks like upstream has each side panel item register itself in the relevant coordinator

Reply:
it looks like this probably needs a follow-up issue?

Reply:
I don't know that we have a similar structure for AIChat. @petemill ?

Reply:
I think we don't need ai chat's coordinator now because what we have to do now is only register it in per-tab registry (`side_panel_registry_`).

Reply:
@simonhong can you open a follow-up issue for the side panel registry related issues?

Reply:
Filed - https://github.com/brave/brave-browser/issues/40949